### DISCUSSION TITLE
Different resize result between output stream and output path image processing

### DISCUSSION THREAD

**Discussion by sekulicb** [saucecontrol/PhotoSauce#64] [Q&A]
Hi. Thank you for this amazing library. I have one interesting problem. While we were testing your library, I had to be sure that the quality and the size of the image will be something desirable, therefor we saved each processed image as **.jpg** file in order to inspect it. For something like this I just used basic example by calling ProcessImage method like this:

`var settings = new ProcessImageSettings()
            {
                Width = 1920,
                Height = 1080,
                ResizeMode = CropScaleMode.Max,
                SaveFormat = FileFormat.Jpeg,
                JpegQuality = 70,
                
                JpegSubsampleMode = ChromaSubsampleMode.Subsample420
            };

            using (var output = new FileStream(Path.Combine(Root, "compressed.jpg"), FileMode.Create))
            {
                await Task.Run(() => MagicImageProcessor.ProcessImage(inputPath, output, settings));
            }`

Image that was ~ 8MB (3264x1836) before processing, is ~ 200KB (1920x1080) after calling MagicImageProcessor which is really great!

Now we had to store these images into the database, which was the whole point of compressing images workflow, to save on load time and space. I have used code that I found here in the following issue: [link here](https://github.com/saucecontrol/PhotoSauce/issues/49#issuecomment-637695415) - keep in mind that I pass identical ProcessImageSettings object, same values no difference.

After using code you suggested, now I get different result. When saving output stream as image in order to test that everything works great again, same image is now ~ 2.8MB which is a lot to store per image, since in our project we load around 600 images every time the app starts. What could be the cause of such huge difference in image processing? To apply quick fix I will just save the image to the output, and then convert the image back to byte[], but there has to be a better option than that. Any thoughts on this? Thanks

**Comment by saucecontrol** [comment]
Howdy.  There are two common mistakes when using Streams.  I'd check these first:

1) When you write the output of an operation to a Stream, the `Position` will point to the end of whatever the output was.  Before reading the data back, you'll need to set `Position = 0`.
2) If you re-use a Stream for multiple operations, you must reset the `Length` of the Stream to 0 between operations, otherwise it will hold on to the previous output's length and ultimately will end up sized to the maximum of any of the outputs.  For example, suppose you process your first image and get an output of 250KB, reset `Position = 0`, then process a second and get an output of 200KB.  The `Length` of the Stream will still be 250KB after the second op, only the first 200KB will be written over, and you'll have 50KB of garbage at the end.
